### DISCUSSION TITLE
Regrading Tensorflow R-cnn resnet and boxes

### DISCUSSION THREAD

**Discussion by MagnusGabell** [jwood803/MLNetExamples#26] [General]
Hi,

I hope you can assist. Dont know if I am in deep waters or if this is simple. I have a tensorflow model faster r-cnn resnet50 that I would like to consume and use in a C# .net6 project. I want to classify objects in images and get boxes and confidence.
Is this something you have done and/or can point to any sources where I can get this. I saw a tutorial about it but it was using ONNX format and I cant change the model. I want to do what you do with inception-model and consume it directly similar to your example but 
with boxes too. Also have the second question in another discussion about the labels. :D

**Comment by jwood803** [comment]
Hi! Yeah, you should be able to use that model using the load and score Tensorflow methods. Is the model available online anywhere? I can try it as well to see what the pipeline should look like.

**Reply by MagnusGabell** [reply]
Hi, thanks for replying. I do not have them online. Its a research project and not public. Can I share a link via email?
The pb file is 140mb :D
I managed, I think, to convert to onnx. It didnt complain when running the script. But I dont know if it will work with the tutorials. 
If it is possible to use the graph straight up, it would be preferred.

I shared the files via your gmail and from onedrive.

**Reply by MagnusGabell** [reply]
python -m tf2onnx.convert --graphdef frozen_inference_graph.pb --output frozen_inference_graph.onnx --fold_const --opset 11 --inputs image_tensor:0 --outputs num_detections:0,detection_boxes:0,detection_scores:0,detection_classes:0

This is what I used to export to ONNX. I get error with the outputs. I need one in the code (output) but I understand 
from the Netron that the outputs are as stated above  num_detections:0,detection_boxes:0,detection_scores:0,detection_classes:0.
I can only add one :-| . Then the input name was image_tensor:0. So I got past the input/out issue when I just then added one of the outputs but the got more errors. BUT if you can get the thing to work with the original model then I will leave the ONNX. That is just a pain.
